conan/cache/cache.py

```diff
@@ -191,9 +147,8 @@ def _move_prev(self, old_pref: ConanReference, new_pref: ConanReference):
 
         return new_path
 
-    def update_recipe(self, old_ref: ConanReference, new_ref: ConanReference = None,
-                         new_path=None, new_timestamp=None):
-        self._db.update_recipe(old_ref, new_ref, new_path, new_timestamp)
+    def update_recipe_timestamp(self, ref: ConanReference,  new_timestamp=None):
```

Comment:
The only scenario of recipe_reference update in the DB is when there is an update of timestamp due to some server updates. Nothing else should/can be updated, so better limit scope.